perses/rjmc/topology_proposal.py

```diff
@@ -742,6 +746,51 @@ def propose(self):
         pass
 
 
+class SystemGenerator(object):
+    """
+    This is a utility class to generate OpenMM Systems from
+    topology objects.
+
+    Parameters
+    ----------
+    forcefields_to_use : list of string
+        List of the names of ffxml files that will be used in system creation.
+    metadata : dict, optional
+        Metadata associated with the SystemGenerator.
+    """
+
+    def __init__(self, forcefields_to_use, metadata=None):
+        self._forcefields = forcefields_to_use
+
+    def build_system(self, new_topology, oemol_list, forcefield_kwargs=None):
```

Comment:
`forcefield_kwargs` should probably go in `__init__`, since you want the generator to cache the forcefields and extra arguments that go to `createSystem`.  You can keep an option for additional `kwargs` here, but we want to make sure that the `SystemGenerator` generates systems in exactly the same way each time.